vnext/Microsoft.ReactNative.Cxx/NativeModules.h

```diff
@@ -301,7 +301,7 @@ constexpr void ValidateCoroutineArg() noexcept {
     static_assert(
         !std::is_reference_v<TArg> && !std::is_pointer_v<TArg>,
         "Coroutine parameter must be passed by value for safe access"
-#ifndef __APPLE__
+#ifndef __clang__
         ": " __FUNCSIG__
```

Comment:
>__FUNCSIG__ [](http://example.com/codeflow?start=13&length=11)

This is MSVC specific macro. I am guilty of adding it and not properly #ifdef-ing it. 
Though instead of using __clang__ we must use a test for MSVC compiler because this code would fail in any other compiler except MSVC.

Reply:
No because _MSC_VER is also defined when compiling with clang-cl.  Internally we've used a combination of __clang__ and MS_TARGET_WINDOWS which doesn't help in this case.  I'd leave it as is until we prove we have consumers of other compilers.

Reply:
Merging as is, then.